prow/plugins/bugzilla/bugzilla.go

```diff
@@ -508,6 +508,23 @@ func validateBug(bug bugzilla.Bug, dependents []bugzilla.Bug, options plugins.Bu
 		}
 	}
 
+	if len(dependents) == 0 {
```

Comment:
This doubles the check -- could we just error if one is not found in the loops above? Also, I expect this change to need an update to the help text

Reply:
> Also, I expect this change to need an update to the help text

You mean [this][1]?  I don't see anything in there about dependent bugs at the moment.  Can't we just rely on the case-specific messages the Bugzilla bot injects when it is unhappy?  Or is there somewhere else that the plugin help is exposed (we don't seem to have it [here][2], and the config is branch-specific anyway, so you'd need a branch-specific location to expose the `helpProvider` output.

[1]: https://github.com/kubernetes/test-infra/blob/ca3411f8f6262cbcee6d5a7700625785e933585a/prow/plugins/bugzilla/bugzilla.go#L52-L150
[2]: https://prow.svc.ci.openshift.org/plugins

Reply:
> This doubles the check -- could we just error if one is not found in the loops above?

By pushing `if len(dependents) == 0` checks into the previous `if options.DependentBugStates != nil` and `if options.DependentBugTargetRelease != nil` blocks?  We could probably do something like that, but it makes it harder to handle the case where both are set, where I want to report:

    expected "+bugLink+" to depend on a bug targeting the %q release and in one of the following states: %s, but no dependents were found

The earlier checks are about "are each of my dependents valid", the check I'm adding is "and if I have no dependents, is that valid?".  Those seem orthogonal enough to me to be handled in separate sections as I have now.